swan-lake/resources/featured-scenarios/build-a-change-data-capture-service-in-ballerina.md

```diff
@@ -0,0 +1,299 @@
+---
+
+## Prerequisites
+
+To follow along with this tutorial, ensure you have:
+
+* [Ballerina 2201.12.0 (Swan Lake)](https://ballerina.io/downloads/) or later
+* A text editor  
+
+  > **Tip:** Preferably [Visual Studio Code](https://code.visualstudio.com/) with the [Ballerina extension](https://marketplace.visualstudio.com/items?itemName=WSO2.ballerina)
+* A terminal or command-line interface
+* MySQL 8.0 or later
+
+## Set up a MySQL server instance
+
+You can use one of the following methods to set up MySQL:
+
+1. **Install MySQL locally** from the [official MySQL site](https://dev.mysql.com/doc/mysql-getting-started/en/#mysql-getting-started-installing)
+2. **Use a server stack** like [XAMPP](https://www.apachefriends.org/) or [WampServer](https://wampserver.aviatechno.net/)
+3. **Run MySQL in Docker** using [these official instructions](https://dev.mysql.com/doc/mysql-installation-excerpt/8.0/en/docker-mysql-getting-started.html)
+4. **Use a cloud-hosted instance** from [Google Cloud SQL](https://cloud.google.com/sql), [Amazon RDS](https://aws.amazon.com/rds/mysql/), or [Azure Database for MySQL](https://azure.microsoft.com/en-us/services/mysql/)
+
+> **Tip:** Keep your MySQL connection details (host, port, username, and password) handy.
+
+## Set up the MySQL database
+
+### Enable CDC in MySQL
+
+CDC relies on MySQL’s binary logging feature, which is enabled by default in MySQL 8.0+ unless explicitly disabled.
+
+#### Verify binary logging
+
+Run the following SQL to confirm:
+
+```sql
+SHOW VARIABLES LIKE 'log_bin';
+```
+
+If `log_bin` is `ON`, you're good to go.
+
+#### Ensure proper CDC settings
+
+Also verify the following:
+
+```sql
+SHOW VARIABLES LIKE 'binlog_format';
+SHOW VARIABLES LIKE 'binlog_row_image';
+```
+
+If needed, update your `my.cnf` or `my.ini` file:
+
+```ini
+[mysqld]
+server-id=223344
+log-bin=mysql-bin
+binlog-format=ROW
+binlog-row-image=FULL
+```
+
+Then restart MySQL to apply the changes.
+
+### Create necessary tables
+
+Log in to your MySQL server and execute:
+
+```sql
+CREATE DATABASE IF NOT EXISTS finance_db;
+USE finance_db;
+
+CREATE TABLE transactions (
+    tx_id INT AUTO_INCREMENT PRIMARY KEY,
+    user_id INT,
+    amount DECIMAL(10,2),
+    status VARCHAR(50),
+    created_at DATETIME
+);
+
+INSERT INTO transactions (user_id, amount, status, created_at) VALUES
+(10, 9000.00, 'COMPLETED', '2025-04-01 08:00:00'),
+(11, 12000.00, 'COMPLETED', '2025-04-01 08:10:00'), -- this triggers fraud logic
+(12, 4500.00, 'PENDING', '2025-04-01 08:30:00');
+```
+
+## Create a Ballerina service package
+
+Ballerina uses packages to group code. You need to create a Ballerina package and write the business logic.
+
+> **Info:** For more information on Ballerina packages, see [Organize Ballerina code](https://ballerina.io/learn/organize-ballerina-code/).
+
+In the terminal, run the following to create a new Ballerina project:
+
+```bash
```

Comment:
```suggestion
```
```

Reply:
Fixed in [098a221](https://github.com/ballerina-platform/ballerina-dev-website/pull/9891/commits/098a221ea98fef64881aca9b621b8c3733ff0ff3)